docs/prover/checking/coverage-info.md

```diff
@@ -0,0 +1,147 @@
+
+* *Are all solidity functions from the input involed in proving my rules?*
+* *Are all solidity commands from the input involed in proving my rules?*
+* *Supposing an `assert` in my rule is not reachable, what is the reason for the unreachability?*
+* *Do I really need all hooks that are defined in my .spec file(s)?*
+* *Do I really need all `require` statements in my rule?*
+* *Do I really need to initialise a CVL variable in my rule?*
+* *Do I really need all preserved blocks in my CVL `invariant`?*
+
+To answer the above questions, the Certora Prover generates a so-called *minimal unsat core* which, intuitively, represents the minimal subset of the commands in the input `.sol` and `.spec` files that are needed to prove the CVL properties. If some of the input `.sol` commands are not needed to derive the proof, it might indicate that the specification does not cover all behavior implemented in the smart contract. If some of the input `.spec` commands are not needed to derive the proof (typically unnecessary `require` statements or variable initializations), it indicates that the CVL rules/invariants can be make stronger. 
+
+We visualize this *coverage* information in a dedicated HTML file: `zipOutput/Reports/UnsatCorevisualization.html`. Furthermore, we also visualize the unsat core coverage information on our `TAC` representation of the verification condition. 
+
+In the rest of this section, we provide a more detailed explanation of the concept of unsat cores and demonstrate it on an example. 
+
+ 
+## Unsat cores
+
+The Certora Prover works as follows:
+1. It takes as an input `.sol` and `.spec` files.
+2. It compiles the `.sol` input into EVM bytecode.
+3. For each single *property*, i.e. elementary rule/invariant, in the `.spec` file(s), it converts the property and the bytecode into a TAC program.
+4. Each TAC program is then converted to an SMT formula such that the formula is unsatisfiable if and only if the property holds (i.e. cannot be violated). 
+
+The SMT formula  is built from a set `A` of *assertions*, say `{A1, A2, ..., An}`, that are built over a set of *variables*. Intuitively, you can see the assertions as mathematical equations. The formula is *satisfiable* if there exists an assignment to the variables that satisfies all the assertions simultaneously. Otherwise, the formula is *unsatisfiable*. In practice, it is often the case that already a small subset of `A` is unsatisfiable. Especially, one can extract a *minimal unsatisfiable subset* `U` of `A`. The *minimality* here means that if you remove any assert from `U` then it becomes satisfiable, i.e., it is not a *minimum cardinality*. We call a minimal unsatisfiable subset of `A` an *unsat core* of `A`. 
+For example, assume that `A = {a, !a, !b, a || b}` where `a` and `b` are Boolean variables. 
+There are two minimal unsatisfiable subsets (i.e. unsat cores) of this formula:
+`{a, !a}` and `{!a, !b, a || b}`.
```

Comment:
I am not sure whether it is a good idea. There are many academic papers I can reference, however, I am afraid it might be confusing (the papers are dealing with particular problems/applications/algorithms that are different than our. 

I think (hope?) that the "definition" I provided is understandable. But I will update the example `A = {a, !a, !b, a || b}` to contain also an assert that is not in any unsat core! E.g. `a || !b`

Reply:
I was thinking more of a script of some lecture or so, or perhaps even a book, if there is one ... (science-y is maybe not the best word ... uhm, didactic, maybe -- university-teaching-y ...)
but yeah, not saying yours isn't rigorous or understandable :-) -- just thought maybe that these too goals at the same time might make things harder than they needed to be here ...